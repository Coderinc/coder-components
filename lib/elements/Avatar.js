import React from 'react';
import styled from 'styled-components';

const AvatarWrapper = styled.div`
  position: relative;

  width: ${props => {
    if (props.large) {return '2rem';}
    else if (props.small) {return '1rem';}
    return '1.5rem';
  }};
  height: ${props => {
    if (props.large) {return '2rem';}
    else if (props.small) {return '1rem';}
    return '1.5rem';
  }};

  border-radius: 50%;
`;

const AvatarImage = styled.img`
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

const AvatarLabel = styled.span`
  position: absolute;
  bottom: -0.25rem;
  left: 50%;
  transform: translate(-50%, 100%);
  padding: 0.25rem;
  border-radius: 0.25rem;

  font-size: 0.625rem;
  line-height: 0.625rem;
  text-align: center;

  color: ${props => props.theme.offsetFont};
  background-color: ${props => props.theme.offsetBackground};
  z-index: -1;
  opacity: 0;

  transition: 0.125s all;
  transition-delay: 0.1s;

  ${AvatarWrapper}:hover & {
    opacity: 1;
    z-index: 5;
  }
`;

export const Avatar = props => (
  <AvatarWrapper
    large={props.large}
    small={props.small}
  >
    <AvatarImage src={props.icon} />
    <AvatarLabel>
      {props.label || `${props.firstName} ${props.lastName}`}
    </AvatarLabel>
  </AvatarWrapper>
);

export default Avatar;
