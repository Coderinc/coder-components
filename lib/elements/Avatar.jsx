import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Tooltip from './Tooltip';

const AvatarWrapper = styled.div`
  width: ${({ large, small }) => {
    if (large) {
      return '2rem';
    }
    if (small) {
      return '1rem';
    }
    return '1.5rem';
  }};
  height: ${({ large, small }) => {
    if (large) {
      return '2rem';
    }
    if (small) {
      return '1rem';
    }
    return '1.5rem';
  }};

  border-radius: 50%;
`;

const AvatarImage = styled.img`
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

const Avatar = ({
  large, small, icon, label, firstName, lastName, ...props
}) => (
  <AvatarWrapper large={large} small={small} {...props}>
    <Tooltip label={label || `${firstName} ${lastName}`}>
      <AvatarImage src={icon} />
    </Tooltip>
  </AvatarWrapper>
);

Avatar.propTypes = {
  large: PropTypes.bool,
  small: PropTypes.bool,
  icon: PropTypes.string.isRequired,
  label: PropTypes.string,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
};

Avatar.defaultProps = {
  large: false,
  small: false,
  label: undefined,
  firstName: 'Joe',
  lastName: 'Schmoe',
};

export default Avatar;
