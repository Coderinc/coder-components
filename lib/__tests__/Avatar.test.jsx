import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../elements/Avatar';

describe('Avatar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Avatar label="test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
