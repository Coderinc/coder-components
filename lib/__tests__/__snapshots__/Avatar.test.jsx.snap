// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="AvatarWrapper-hhMvHp ekaaMQ"
>
  <div
    className="TooltipWrapper-kssXFQ ggKbDb"
  >
    <span
      className="TooltipLabel-dIPTMH ePYwYi"
    >
      test
    </span>
    <div
      className="TooltipArrow-fsUgpx jktQis"
    />
    <img
      alt="test"
      className="AvatarImage-cTHrxV ednqDG"
      src={null}
    />
  </div>
</div>
`;
